From 5596617baacbe99f0c4dc6d97417d16b6b1b69d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Thu, 6 Apr 2023 22:14:48 +0200
Subject: [PATCH] Do not install documentation

---
 Makefile.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.rules b/Makefile.rules
index 1b8b4d2c..befbec0c 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -570,7 +570,7 @@ clean:
 
 LTINSTALL=$(LIBTOOL) --mode=install $(INSTALL)
 
-install: install-bin install-doc install-gaproot install-sysinfo install-headers install-libgap
+install: install-bin install-gaproot install-sysinfo install-headers install-libgap
 	@echo "+--------------------------------------------------------------------------+"
 	@echo "| WARNING, 'make install' support is still experimental, and may be buggy. |"
 	@echo "| It also does not take care of installing GAP packages (and may never).   |"
-- 
2.20.1

